src/Tools/ExternalAccess/FSharp/Internal/Diagnostics/FSharpUnusedDeclarationsAnalyzer.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.ExternalAccess.FSharp.Diagnostics;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Host.Mef;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.FSharp.Internal.Diagnostics
+{
+    [Shared]
+    [ExportLanguageService(typeof(FSharpUnusedDeclarationsDiagnosticAnalyzerService), LanguageNames.FSharp)]
+    internal class FSharpUnusedDeclarationsDiagnosticAnalyzerService : ILanguageService
+    {
+        private readonly IFSharpUnusedDeclarationsDiagnosticAnalyzer _analyzer;
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public FSharpUnusedDeclarationsDiagnosticAnalyzerService(IFSharpUnusedDeclarationsDiagnosticAnalyzer analyzer)
+        {
+            _analyzer = analyzer;
+        }
+
+        public Task<ImmutableArray<Diagnostic>> AnalyzeSemanticsAsync(DiagnosticDescriptor descriptor, Document document, CancellationToken cancellationToken)
+        {
+            return _analyzer.AnalyzeSemanticsAsync(descriptor, document, cancellationToken);
+        }
+    }
+
+    [DiagnosticAnalyzer(LanguageNames.FSharp)]
+    internal class FSharpUnusedDeclarationsDiagnosticAnalyzer : DocumentDiagnosticAnalyzer, IBuiltInAnalyzer
```

Comment:
💡 `IBuiltInAnalyzer` should not be used

```suggestion
    internal class FSharpUnusedDeclarationsDiagnosticAnalyzer : DocumentDiagnosticAnalyzer
```

Reply:
Can you expand on this? I'm not sure why it would not be used.

Reply:
We are working to eliminate it. `DocumentDiagnosticAnalyzer` has special handling anyway which should supersede `IBuiltInAnalyzer`.